src/viewport/viewport.js

```diff
@@ -64,17 +64,36 @@ export default class WebGLViewport extends Viewport {
     // TODO: move the following line to initialization so that it's done only once
     const positionOriginPixels = this.projectFlat(positionOrigin);
 
-    const delta = vec2.subtract([],
-       positionOriginPixels,
-       this.center
-     );
-
-    const projectionCenter = vec2.add([],
-      this.center,
-      delta);
-
+    const projectedPositionOrigin = [positionOriginPixels[0], positionOriginPixels[1], 0.0, 1.0];
     const projections = this.getProjections();
-    const projectionMatrix = projections._viewProjectionMatrix;
+    const projectionMatrix = projections.viewProjectionMatrix;
+
+    const projectionCenter = [0.0, 0.0, 0.0, 0.0];
+
+    projectionCenter[0] =
```

Comment:
`vec4` has a `transformMat4` method. Documented towards the end of http://glmatrix.net/docs/vec4.html.